tests/api.spec.ts

```diff
@@ -194,20 +194,24 @@ describe('Requests to public API', () => {
   });
 
   it('request analysis with missing files', async () => {
-    const response = await getAnalysis({
-      baseURL,
-      sessionToken,
-      bundleId: fakeBundleIdFull,
-      includeLint: false,
-      severity: 1,
-    });
+    let response;
+    do {
+      response = await getAnalysis({
+        baseURL,
+        sessionToken,
+        bundleId: fakeBundleIdFull,
+        includeLint: false,
+        severity: 1,
+      });
+    } while (response.type === 'success');
+
     expect(response.type).toEqual('error');
-    if (response.type === 'success') return;
     expect(response.error).toEqual({
       apiName: 'getAnalysis',
       statusCode: 500,
```

Comment:
Shouldn't this be 404 instead of 500 for incomplete/expired bundle analysis in the new pubAPI queue branch?

Reply:
hm, this is interesting question. let me check...

Reply:
fixed. Please check with together with this PR: https://github.com/DeepCodeAI/product/pull/643